### ISSUE TITLE
Wrong UMD ID's of Angular packages with a dash in name

### ISSUE THREAD

**Issue by flash-me** [angular/angular#35765]
# bug report

### Affected Package
Every angular package with a dash in their names.
So far:
- "@angular/cdk/drag-drop"
- "@angular/cdk/text-field"
- "@angular/material/form-field"
- "@angular/material/button-toggle"
- "@angular/material/snack-bar"
- "@angular/material/bottom-sheet"
- "@angular/material/grid-list"
- "@angular/material/progress-bar"
- "@angular/material/progress-spinner"
- "@angular/material/slide-toggle"

### Is this a regression?
At least it requires a workaround to function.

### Description

 A clear and concise description of the problem...

These packages are published with a UMD ID by camelCasing their names, for e.g.  <br>
`@angular/cdk/drag-drop` results of other packages looking for `ng.dragDrop` instead of `ng.material['drag-drop']`.

When providing the UMD bundles of these packages by bundling them with ng-packagr, this leads to unexpected behaviour, since ng-packagr will provide them globally with leaving their dashes in their names (like `ng.material['drag-drop']`). But since other packages of the angular framework are looking for the camelCase version, some manual work has to be done to make this function.

So far, I make use of the `umdModuleIds` property of the [ng-packagr.schema.json](https://github.com/ng-packagr/ng-packagr/blob/master/src/ng-package.schema.json#L57) to make this work, so my config looks like this:
```
"umdModuleIds": {
	"@angular/cdk/drag-drop": "ng.cdk.dragDrop",
	"@angular/cdk/text-field": "ng.cdk.textField",
	"@angular/material/form-field": "ng.material.formField",
	"@angular/material/button-toggle": "ng.material.buttonToggle",
	"@angular/material/snack-bar": "ng.material.snackBar",
	"@angular/material/bottom-sheet": "ng.material.bottomSheet",
	"@angular/material/grid-list": "ng.material.gridList",
	"@angular/material/progress-bar": "ng.material.progressBar",
	"@angular/material/progress-spinner": "ng.material.progressSpinner",
	"@angular/material/slide-toggle": "ng.material.slideToggle"
}
```
so the other packages will find the according UMD ID's

## 🔬 Minimal Reproduction

Just have a look in the according released npm packages...

## 🌍  Your Environment

**Angular Version:** Latest Greatest (stable)
<pre><code>
Angular CLI: 9.0.3
Node: 12.16.1
OS: linux x64

Angular: 9.0.3
... animations, cli, common, compiler, compiler-cli, core
... elements, forms, platform-browser
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.900.3
@angular-devkit/build-angular      0.900.3
@angular-devkit/build-ng-packagr   0.900.3
@angular-devkit/build-optimizer    0.900.3
@angular-devkit/build-webpack      0.900.3
@angular-devkit/core               9.0.3
@angular-devkit/schematics         9.0.3
@angular-devkit/schematics-cli     0.900.4
@angular/cdk                       9.1.0
@angular/material                  9.1.0
@ngtools/webpack                   9.0.3
@schematics/angular                9.0.3
@schematics/schematics             0.900.4
@schematics/update                 0.900.3
ng-packagr                         9.0.2
rxjs                               6.5.3
typescript                         3.7.4
webpack                            4.41.2

</code></pre>

**Anything else relevant?**

It would be nice if a unified approach can be taken. Either camelCase everywhere, or kebab-case everywhere

cheers
flash :zap:

**Comment by flash-me** [comment]
Just asked myself why this works with `global.ng.platformBrowser`. Maybe not even a problem of @angular but ng-packagr?

**Comment by atscott** [comment]
I'm sorry but this issue is not caused by Angular. Please contact the author(s) of project https://github.com/angular/components or file issue on their issue tracker.

**Comment by flash-me** [comment]
Hi @atscott,

did you even read the complete issue or did you just see the material and cdk names in the snippet before you closed this?

> Every angular package with a dash in their names

In which world are
- `@angular/platform-browser` or
- `@angular/platform-browser-dynamic`

NOT part of @angular?

Anyway. Have a nice weekend dude

Cheers
flash :zap:

**Comment by petebacondarwin** [comment]
@flash-me - sorry if this seems like it was closed too quickly.

In this project (angular/angular) we use (rollup via the `ng_package` Bazel rule) to  generate the bundles, which is configured to use camelCase within this repository: https://github.com/angular/angular/blob/93c2df23bfd893fd0ff8c5e9c03b332b9b8e7124/packages/bazel/src/ng_package/ng_package.bzl#L260-L264

I guess the question is whether this is necessary; and if not whether it would be too much of a breaking change to modify it now.

**Comment by flash-me** [comment]
@petebacondarwin It's okay. Somewhere it's reasonable to be able to handle all the incoming issues. And sorry to @atscott for being harsh at first! Kinda overreacted.

B2T:
 `camelCase` would be totally okay, if used everyhwere. Would it be possible to configure rollup to provide dashed UMD ID's also in `camelCase`? e.g.

So instead of providing as `ng['platform-browser']` to provide it as `ng.platformBrowser`

 This would solve the problem. The problem is just the different approach in how they are provided at bundle time and required at runtime.

Cheers
flash :zap:

**Comment by alan-agius4** [comment]
@flash-me, `ng-packagr` tries to set defaults module id's for `@angular` packages, however it seems that for some cases this is not done correctly.

Thus, this is a bug in ng-packagr, Here's a PR to address this issue:

**Comment by alan-agius4** [comment]
@flash-me should be fixed in ng-packagr version `9.0.3`.

**Comment by flash-me** [comment]
Hi @alan-agius4

just tried out. Here are the results:

**Before**
![before](https://user-images.githubusercontent.com/5240450/76226082-a9c42880-621d-11ea-9266-993930e6d7e8.png)
**After**
![after](https://user-images.githubusercontent.com/5240450/76226123-b5afea80-621d-11ea-8232-b23af048214f.png)

Seems to be resolved :+1: 
Thanks for the fast response!

cheers
flash :zap:
